jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/handler/VariableResolver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.bpmn2.handler;
+
+import org.jbpm.process.core.event.EventTransformerImpl;
+import org.jbpm.workflow.core.node.Transformation;
+import org.kie.api.runtime.process.ProcessContext;
+
+class VariableResolver {
+
+    private VariableResolver() {}
+
+    public static Object getVariable(ProcessContext kcontext, String varName) {
+        Object tVariable = null;
```

Comment:
this logic should be in the incoming input association. you are spreading the logic.
Just add one item to that (you have the variable plus the transformation it is easy to build in there instead of here).

Reply:
since this is just a copy of the current functionality in drools action (which have been translated to Java), that change is out of the scope of this JIRA and will be handled with a different one.